presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java

```diff
@@ -217,76 +226,20 @@ public boolean isDone()
         return taskContext.isDone();
     }
 
-    public void transferMemoryToTaskContext(long bytes)
-    {
-        // The memory is already reserved in the task context, so just decrement our reservation
-        checkArgument(memoryReservation.addAndGet(-bytes) >= 0, "Tried to transfer more memory than is reserved");
-    }
-
-    public synchronized ListenableFuture<?> reserveMemory(long bytes)
-    {
-        ListenableFuture<?> future = taskContext.reserveMemory(bytes);
-        memoryReservation.getAndAdd(bytes);
-        return future;
-    }
-
-    public synchronized ListenableFuture<?> reserveRevocableMemory(long bytes)
-    {
-        ListenableFuture<?> future = taskContext.reserveRevocableMemory(bytes);
-        revocableMemoryReservation.getAndAdd(bytes);
-        return future;
-    }
-
-    public synchronized ListenableFuture<?> reserveSystemMemory(long bytes)
-    {
-        checkArgument(bytes >= 0, "bytes is negative");
-        ListenableFuture<?> future = taskContext.reserveSystemMemory(bytes);
-        systemMemoryReservation.getAndAdd(bytes);
-        return future;
-    }
-
     public synchronized ListenableFuture<?> reserveSpill(long bytes)
     {
         return taskContext.reserveSpill(bytes);
     }
 
-    public synchronized boolean tryReserveMemory(long bytes)
-    {
-        if (taskContext.tryReserveMemory(bytes)) {
-            memoryReservation.getAndAdd(bytes);
-            return true;
-        }
-        return false;
-    }
-
-    public synchronized void freeMemory(long bytes)
-    {
-        checkArgument(bytes >= 0, "bytes is negative");
-        checkArgument(bytes <= memoryReservation.get(), "tried to free more memory than is reserved");
-        taskContext.freeMemory(bytes);
-        memoryReservation.getAndAdd(-bytes);
-    }
-
-    public synchronized void freeRevocableMemory(long bytes)
-    {
-        checkArgument(bytes >= 0, "bytes is negative");
-        checkArgument(bytes <= revocableMemoryReservation.get(), "tried to free more revocable memory than is reserved");
-        taskContext.freeRevocableMemory(bytes);
-        revocableMemoryReservation.getAndAdd(-bytes);
-    }
-
-    public synchronized void freeSystemMemory(long bytes)
+    public synchronized void freeSpill(long bytes)
     {
         checkArgument(bytes >= 0, "bytes is negative");
-        checkArgument(bytes <= systemMemoryReservation.get(), "tried to free more system memory than is reserved");
-        taskContext.freeSystemMemory(bytes);
-        systemMemoryReservation.getAndAdd(-bytes);
+        taskContext.freeSpill(bytes);
     }
 
-    public synchronized void freeSpill(long bytes)
+    public LocalMemoryContext localSystemMemoryContext()
     {
-        checkArgument(bytes >= 0, "bytes is negative");
-        taskContext.freeSpill(bytes);
+        return pipelineMemoryContext.localSystemMemoryContext();
```

Comment:
You declared `GuardedBy` for `pipelineMemoryContext`. But it was never used inside guards.